### PR TITLE
Pathlib vs OS Modules for File and Directory System Operation utilizing Comparable Scenarios

### PR DESCRIPTION
**Pull request by akenz1901**
## Our **Peer Reviewer and Content Moderator teams do NOT provide any revisions services.** All revisions and edits should be completed on your own forked repo (as to not take up room in the queue) in order for our team to review them all in a timely manner. 

**ANY ARTICLE SUBMITTED WITH GLARING ERRORS WILL BE IMMEDIATELY CLOSED.**

>As a rule of thumb - please be sure to only submit articles (pull requests) that are fully polished and ready to be published. Be sure to go through our resources documents for extra and 3rd party (vetted) resources to help improve overall technical writing.

> If you have not already, please go over our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for more tips and tricks on how to improve your overall technical writing so reviews are swifter, increase the chance of a payout and provide more value to our readers.

> For first-time contributors and for more details on our submission guidelines, **see our [Contributing Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**.

#### Formatting and structure checks
- [ ] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [ ] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [ ] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [ ] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**

- [ ] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [ ] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites. See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [ ] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [ ] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**

- [ ] Have you included a hero image that is **600x400 pixels**, under **300KB in size**, and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
      
#### Grammar & plagiarism checks
- [ ] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) - [Grammarly](https://grammarly.com) 
- [ ] Please place your article through a 3rd party plagiarism checker. We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [ ] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [ ] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [ ] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**
- [ ] Have you checked your code runs correctly and you've highlighted all necessary dependencies for installation?
- [ ] Are the software programs and packages you're highlighting in your article up to date, using current versions and not deprecated?

`This closes #issue-number`

> Finally, delete the article checklist notes in blockquotes and submit your PR. We look forward to reviewing your article.

### CODE REVIEW

**Comment by srishilesh** [comment]
Duplicate of #6145

**Comment by srishilesh** [comment]
I'll complete my tech review by tomorrow EOD.

**Comment by akenz1901** [comment]
@srishilesh 
Thanks so much, I will be expecting your feeback.

File: content/articles/pathlib-vs-os-modules-for-file-and-directory-system-operation/index.md [hunk 1]
```diff
@@ -0,0 +1,910 @@
...
+- [Querying path to retrieve a final file or directory name](#querying-path-to-retrieve-a-final-file-or-directory-name)
+- [Removing a directory](#removing-a-directory)
+- [Removing a file](#removing-a-file)
+- [Retrieving and converting a file information](#retrieving-and-converting-a-file-information)
+- [Querying a file extension and name](#querying-a-file-extension-and-name)
+- [Theoretical difference between pathlib and Os](#theoretical-difference-between-pathlib-and-os)
+- [Wrapping Up](#wrapping-up)
+
+### Introduction
+File System Operation can be characterized as the example or technique that can be used to control or construct files and directories and there are two different ways of working with file system operation, which are functional dependency and object-oriented concept. This tutorial practically highlights the different scenarios to differentiate between Pathlib and Os.
```
Comment (srishilesh) [inline content/articles/pathlib-vs-os-modules-for-file-and-directory-system-operation/index.md:40]: Make sure you maintain consistency while mentioning the name of the libraries - `Pathlib` and `OS`.

Do this change throughout the article

File: content/articles/pathlib-vs-os-modules-for-file-and-directory-system-operation/index.md [hunk 2]
```diff
@@ -0,0 +1,910 @@
...
+
+#### Os module
+The OS (operating system) module is a different operating system interface that functionally renders a convenient way of working with files and directories with a simple function call, additionally, OS empowers us to deal with I/O tasks straightforwardly.
```
Comment (srishilesh) [inline content/articles/pathlib-vs-os-modules-for-file-and-directory-system-operation/index.md:43]: Add link to the library's website

File: content/articles/pathlib-vs-os-modules-for-file-and-directory-system-operation/index.md [hunk 3]
```diff
@@ -0,0 +1,910 @@
...
+
+#### Pathlib module
+The Pathlib module is an Object-Oriented file system activity that offers classes representing filesystem paths with semantics fitting for various operating systems.
```
Comment (srishilesh) [inline content/articles/pathlib-vs-os-modules-for-file-and-directory-system-operation/index.md:46]: Add link to the library's website

File: content/articles/pathlib-vs-os-modules-for-file-and-directory-system-operation/index.md [hunk 4]
```diff
@@ -0,0 +1,910 @@
...
+
+### Highlighting the practical difference between os and pathlib modules
+In this section we will practically illustrate the semantic differences between pathlib and os modules using comparable code snippets to help you determine where every module fit in. Additionally, note that every demonstration requires the importation of the two modules.
+
+### Querying current path using os and pathlib
```
Comment (srishilesh) [inline content/articles/pathlib-vs-os-modules-for-file-and-directory-system-operation/index.md:51]: `###` -> `####`

Do this change for the below sub-topics

Comment (srishilesh) [reply content/articles/pathlib-vs-os-modules-for-file-and-directory-system-operation/index.md:51]: `OS and Pathlib`

File: content/articles/pathlib-vs-os-modules-for-file-and-directory-system-operation/index.md [hunk 5]
```diff
@@ -0,0 +1,910 @@
...
+Querying the current working directory is a basic task when you are dealing with a path.  There are two unique types of path which are relative path and absolute path.An absolute path gives a full description of the current location you are working while a relative path refers to a location that is relative to the current working directory. Note querying the current path using os and pathlib will return an absolute path.
```
Comment (srishilesh) [inline content/articles/pathlib-vs-os-modules-for-file-and-directory-system-operation/index.md:52]: Inconsistent spacing after and before full-stops

File: content/articles/pathlib-vs-os-modules-for-file-and-directory-system-operation/index.md [hunk 6]
```diff
@@ -0,0 +1,910 @@
...
+ WindowsPath('../WPS Office.lnk'),
+ WindowsPath('../WPS PDF.lnk')]
+```
+The output is the list of the files and directories in the prompt parent directory which tends to be my Desktop directory.
+
+However, pathlib alternatively provides `Path.glob()` to list files and directories in a given or current directory, and this method requires a string argument to perform a specific task.
+
+Consider the following example:
+
+#### Utilizing glob method to outline a relative directory path
```
Comment (srishilesh) [inline content/articles/pathlib-vs-os-modules-for-file-and-directory-system-operation/index.md:258]: What does a glob method do?

File: content/articles/pathlib-vs-os-modules-for-file-and-directory-system-operation/index.md [hunk 7]
```diff
@@ -0,0 +1,910 @@
...
+ WindowsPath('../Visual Studio Code.lnk'),
+ WindowsPath('../WPS Office.lnk'),
+ WindowsPath('../WPS PDF.lnk')]
+```
+It returns all the files that end with the given pattern pathlib module also provides `PurePath.suffix()`method for performing the same operation without having to necessarily convert it to a list.
+
+### Creating a directory and file
+Basically you can create a directory or file using the option provided by your operating system, However in some situations you may need to automatically create a directory or file whenever you are writing a python script, in that case, we will be utilizing this opportunity to semantically illustrate how we can create a file or directory in python.
+
+#### Creating a file using os module.
```
Comment (srishilesh) [inline content/articles/pathlib-vs-os-modules-for-file-and-directory-system-operation/index.md:320]: Remove `.`

File: content/articles/pathlib-vs-os-modules-for-file-and-directory-system-operation/index.md [hunk 8]
```diff
@@ -0,0 +1,910 @@
...
+
+```python
+import os
+
+note_file = os.stat('Untitled.ipynb')
+
+print(f'Size = {note_file.st_size}\nlast time accessed = {note_file.st_atime}\nlast time modified = {note_file.st_mtime}\ncreation time = {note_file.st_ctime}')
+
+```
+In the snippet above we assigned the returned value to the variable named `note_file`, note the function `os.stat()` returns an `os.stat_result` object which provides several public attributes that can be used to retrieve a given file's information. We are only utilizing the corresponding attributes in this section which are `st_size`, `st_atime`, `st_mtime`, `st_ctime`. The `st_size` contains the file size in bytes, the `st_atime` contains the date of the most recent access to the given file in seconds, the `st_mtime` attribute contains the file last modified date in seconds, the `st_ctime` attribute respond differently on Unix and Windows operating system, for windows it contains the creation date, as for Unix it contains the changed date, either you can use `st_birthtime` for querying creation time on Unix operating system.
```
Comment (srishilesh) [inline content/articles/pathlib-vs-os-modules-for-file-and-directory-system-operation/index.md:719]: Break down this para into smaller paras

File: content/authors/akenz-michael/index.md [hunk 9]
```diff
@@ -0,0 +1,12 @@
...
+type: authors
+linkedin: https://www.linkedin.com/in/akinsanya-m-0585661ab/
+email: akenz1901@gmail.com
+github: https://www.github.com/akenz1901
+images:
+- url: /engineering-education/authors/akenz-michael/avatar.jpg
+---
+Akinsanya Michael is a passionate Software Engineer, an ethnographer, a Farmer, and an Industrial Robotics enthusiast,
+he holds a BSc degree in Computer Science and he solves problems with the following tools Ethnography, Java, Python, and SQL.
+He believes sharing and utilizing his skills and knowledge is a great opportunity for achieving growth development.
```
Comment (srishilesh) [inline content/authors/akenz-michael/index.md:12]:
```suggestion
Akinsanya Michael is a passionate Software Engineer, an ethnographer, a Farmer, and an Industrial Robotics enthusiast. He holds a B.Sc degree in Computer Science and he solves problems with the following tools Ethnography, Java, Python, and SQL. He believes sharing and utilizing his skills and knowledge is a great opportunity for achieving growth development.
```

**Review by srishilesh — CHANGES_REQUESTED** [review]
Hey @akenz1901 ,

On performing a review, I noticed very minor changes in formatting and consistencies of `OS` and `Pathlib` library usage. I believe it will take very less time to address these minor suggestions.

Ping me when done.

Thanks,

**Comment by srishilesh** [comment]
@akenz1901 

Also, can you link this PR to an issue? I don't find the issue number.

File: content/articles/pathlib-vs-os-modules-for-file-and-directory-system-operation/index.md (hunk 2 above)
Comment (akenz1901) [reply content/articles/pathlib-vs-os-modules-for-file-and-directory-system-operation/index.md:43]: Alright i will

File: content/articles/pathlib-vs-os-modules-for-file-and-directory-system-operation/index.md (hunk 3 above)
Comment (akenz1901) [reply content/articles/pathlib-vs-os-modules-for-file-and-directory-system-operation/index.md:46]: Thank you so much I will do so now.

File: content/articles/pathlib-vs-os-modules-for-file-and-directory-system-operation/index.md (hunk 4 above)
Comment (akenz1901) [reply content/articles/pathlib-vs-os-modules-for-file-and-directory-system-operation/index.md:51]: Okay i will do so.

File: content/articles/pathlib-vs-os-modules-for-file-and-directory-system-operation/index.md (hunk 5 above)
Comment (akenz1901) [reply content/articles/pathlib-vs-os-modules-for-file-and-directory-system-operation/index.md:52]: Alright do you mean i should apply enough spacing after the full-stop sir?

Comment (akenz1901) [reply content/articles/pathlib-vs-os-modules-for-file-and-directory-system-operation/index.md:52]: Alright, do you mean I should apply enough spacing after the full-stop sir?

File: content/articles/pathlib-vs-os-modules-for-file-and-directory-system-operation/index.md (hunk 7 above)
Comment (akenz1901) [reply content/articles/pathlib-vs-os-modules-for-file-and-directory-system-operation/index.md:320]: Okay

File: content/articles/pathlib-vs-os-modules-for-file-and-directory-system-operation/index.md (hunk 5 above)
Comment (srishilesh) [reply content/articles/pathlib-vs-os-modules-for-file-and-directory-system-operation/index.md:52]: There must be a single space before a full-stop @akenz1901

**Comment by hectorkambow** [comment]
Good afternoon and thank you for submitting your topic suggestion. Not a problem but it seems that this PR is still quite a bit aways from being completed.

In an effort to be fair to our larger EngEd community as a rule of thumb - we can only accept content (pull requests) that are complete and fully polished articles/tutorial - meaning they are ready to be published.

All revisions and edits should be completed on your own forked repo (as to not take up room in the queue) - this allows all revisions to be done on your forked personal repo.

@akenz1901

**Comment by akenz1901** [comment]
Thanks, @hectorkambow , I so much understand your point and i appreciate your the advice.

**Comment by akenz1901** [comment]
Thanks, @hectorkambow, I so much understand your point and I appreciate your advice.

**Pull request closed without merging**